### PR TITLE
Even more scoop up

### PR DESCRIPTION
**Pull request by flagrama**
Even some Gen4 stuff because I'm sick of this refactor and never want to have to look at scoop up again :laughing:

### CODE REVIEW

**Comment by flagrama** [comment]
[MR_MIME_66_TEAM_UP](https://trello.com/c/m4I95Cq4/647-mrmime66teamup)

File: src/tcgwars/logic/groovy/TcgStatics.groovy [hunk 1]
```diff
@@ -1577,6 +1579,7 @@ class TcgStatics {
       target.owner.pbg.discard.addAll(toDiscard2)
       toDiscard.getExcludedList(toDiscard2).discard()
       removePCS(target)
+      return true
     }
   }
```
Comment (starg09) [inline src/tcgwars/logic/groovy/TcgStatics.groovy:1584]: Not sure if relevant, but wouldn't a `return false` be needed after the targeted, in case it gets blocked?

File: src/tcgwars/logic/impl/gen4/DiamondPearlPromos.groovy [hunk 2]
```diff
@@ -629,6 +630,13 @@ public enum DiamondPearlPromos implements LogicCardInfo {
           pokePower "Leap Away", {
             text "Once during your turn , if Toxicroak is your Active Pok�mon, you may flip a coin. If heads, return Toxicroak and all cards attached to it to your hand. This power can�t be used if Toxicroak is affected by a Special Condition."
             actionA {
+              assert self.active : "$self is not your Active Pokémon"
+              checkLastTurn()
+              checkNoSPC()
+              powerUsed()
```
Comment (starg09) [inline src/tcgwars/logic/impl/gen4/DiamondPearlPromos.groovy:636]: Minor suggestion, but since implementing: Would maybe add a confirm when it's the only pokemon in play before using the power, similar to Super Scoop Up.

Might be worth its own PR though, maybe adding a static assert for all cards like this, that checks for no bench and in that case does the confirm.

File: src/tcgwars/logic/impl/gen4/HeartgoldSoulsilver.groovy [hunk 3]
```diff
@@ -274,8 +274,7 @@ public enum HeartgoldSoulsilver implements LogicCardInfo {
             }
```
Comment (starg09) [inline src/tcgwars/logic/impl/gen4/HeartgoldSoulsilver.groovy:274]: Line above, should be usable with no bench. See: review comment on Toxicroak.

File: src/tcgwars/logic/impl/gen4/Platinum.groovy [hunk 4]
```diff
@@ -2660,9 +2663,13 @@ public enum Platinum implements LogicCardInfo {
       case POKE_TURN_118:
         return itemCard (this) {
           text "Return 1 of your Pokémon SP and all cards attached to it to your hand."
+          // TODO: Check for Pokémon SP in onPlay and playRequirement
           onPlay {
+            def pcs = my.bench.findAll { true }.select("Which Pokémon SP to return to hand?")
+            scoopUpPokemon(pcs, delegate)
           }
           playRequirement{
+            my.bench.any { true }
```
Comment (starg09) [inline src/tcgwars/logic/impl/gen4/Platinum.groovy:2672]: Can be played on your only Pokémon in play. See: Super Scoop Up, review comment on Toxicroak.

File: src/tcgwars/logic/groovy/TcgStatics.groovy (hunk 1 above)
Comment (flagrama) [reply src/tcgwars/logic/groovy/TcgStatics.groovy:1584]: Default value of a boolean is false, so if it reaches the end of the method it returns false. I assume the return inside the targeted will not be run if targeted is blocked. Is that incorrect?

File: src/tcgwars/logic/impl/gen4/DiamondPearlPromos.groovy (hunk 2 above)
Comment (flagrama) [reply src/tcgwars/logic/impl/gen4/DiamondPearlPromos.groovy:636]: I'd prefer just getting these to work "good enough" for testing purposes. Primarily just don't want to have to review literally every Gen 4 PR just in case it needs to use scoop up.

File: src/tcgwars/logic/groovy/TcgStatics.groovy (hunk 1 above)
Comment (flagrama) [reply src/tcgwars/logic/groovy/TcgStatics.groovy]: I did it just in case either way.

File: src/tcgwars/logic/impl/gen4/DiamondPearlPromos.groovy (hunk 2 above)
Comment (flagrama) [reply src/tcgwars/logic/impl/gen4/DiamondPearlPromos.groovy:636]: The rest of these should all be part of said scoop-up check PR. The existing ones, and the ones I added, avoid the issue by just not allowing the move be used. I don't think it's closely related enough to this PR to handle here.

**Review by starg09 — APPROVED** [review]
